pkg/apprepo/apprepos_handler.go

```diff
@@ -248,6 +252,16 @@ func (a *AppRepositoriesHandler) DeleteAppRepository(repoName, repoNamespace, to
 	return err
 }
 
+// GetAppRepository returns an AppRepository resource from a namespace.
+// Optionally set a token to get the AppRepository using a custom serviceaccount
+func (a *AppRepositoriesHandler) GetAppRepository(repoName, repoNamespace, token string) (*v1alpha1.AppRepository, error) {
+	clientset, err := a.clientsetForRequest(token)
```

Comment:
Right, this looks like we're depending on `clientsetForRequest("")` returning the service account clientset. I'm -1 on that, it's too easy to mistakenly end up with a service account client set. I think we should remove that and do something like:
```
if token != "" {
    clientset, err := a.clientsetForToken(token)
} else {
    clientset = svcKubeClient
}
```

In fact, even then the optional token on this public method is very implicit at call-sites. We could additionally make this fn private (`getAppRepository`) and instead expose two very small (single-line) wrappers:
```
func (a *AppRepositoriesHandler) GetAppRepositoryForUser(repoName, repoNamespace, token string) ...
```
and 
```
func (a *AppRepositoriesHandler) GetAppRepositoryForServiceAccount(repoName, repoNamespace string) ...
```
calling the private method.